block-producer/src/txqueue/mod.rs

```diff
@@ -0,0 +1,132 @@
+use std::{fmt::Debug, sync::Arc, time::Duration};
+
+use async_trait::async_trait;
+use tokio::{sync::RwLock, time};
+
+use crate::{batch_builder::BatchBuilder, SharedProvenTx, SharedRwVec};
+
+// TRANSACTION QUEUE
+// ================================================================================================
+
+#[derive(Debug)]
+pub enum VerifyTxError {}
+
+#[async_trait]
+pub trait TransactionVerifier: Send + Sync + 'static {
+    async fn verify_tx(
+        &self,
+        tx: SharedProvenTx,
+    ) -> Result<(), VerifyTxError>;
+}
+
+#[derive(Debug)]
+pub enum AddTransactionError {
+    VerificationFailed(VerifyTxError),
+}
+
+#[async_trait]
+pub trait TransactionQueue: Send + Sync + 'static {
+    async fn add_transaction(
+        &self,
+        tx: SharedProvenTx,
+    ) -> Result<(), AddTransactionError>;
+}
+
+// DEFAULT TRANSACTION QUEUE
+// ================================================================================================
+
+pub struct DefaultTransactionQueueOptions {
+    /// The frequency at which we try to build batches from transactions in the queue
+    pub build_batch_frequency: Duration,
+
+    /// The size of a batch
+    pub batch_size: usize,
+}
+
+pub struct DefaultTransactionQueue<BB: BatchBuilder, TV: TransactionVerifier> {
```

Comment:
My preference is to follow what is standard instead of having special rules. Everyone has their personal preferences and these things  rarely have the right way of doing it :trademark: .

I have found the same issue that you mentioned, and also the issue of having the constraints "infecting" unrelated code. Neither are big issues.

With that said I don't mind either, but I would like to have the preferred way documented, and used across all our repos.